### PR TITLE
803 explain path aws subnet new

### PR DESCRIPTION
**Pull request by ShiriMoran**
private subnet added to explain (path, header explaination and detailed explaination)
One test added and verified
more tests yet needs to be added (https://github.com/np-guard/vpc-network-config-analyzer/issues/756)

### CODE REVIEW

**Review by haim-kermany — CHANGES_REQUESTED** [review]
consider code reorg of explainabilityPrint.go

**Comment by ShiriMoran** [comment]
more examples of explain + aws private subnets needs to be added

> consider code reorg of explainabilityPrint.go

Feel free to open an issue and elaborate more in the issue

File: pkg/awsvpc/examples/out/explain_out/to_external_private_subnet_all_vpcs_explain_detail.txt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+====================================================================
+
+No connections from app1[10.240.20.245] to Public Internet 161.26.0.0/32;
+connection is blocked by egress
+
+External traffic via InternetGateway: internet_gw
+Egress: private subnet application denies connection; security group GroupName:43 blocks connection; network ACL NetworkAclId:65 allows connection
+
+Path:
+	app1[10.240.20.245] -> | security group GroupName:43 |
```
[the remark this answers is not in the document]
Comment (ShiriMoran) [reply pkg/awsvpc/examples/out/explain_out/to_external_private_subnet_all_vpcs_explain_detail.txt:13]: 1. The SG indeed block egress. I verified against the json file
2. For egress, SG is before the subnet see [here](https://github.com/np-guard/vpc-network-config-analyzer/issues/769)

File: pkg/ibmvpc/vpc.go [hunk 2]
```diff
@@ -223,7 +223,13 @@ func NewLoadBalancerRule(lb *LoadBalancer, deny bool, src, dst vpcmodel.Node) vp
 
 func (lbr *LoadBalancerRule) Deny(isIngress bool) bool { return !isIngress && lbr.deny }
 
-func (lbr *LoadBalancerRule) String() string {
+// IsIngress load balancer potentially blocks egress connection
+func (lbr *LoadBalancerRule) IsIngress() bool {
+	return false
+}
+
+// todo: use detail to get a concise printing for !detail
```
[the remark this answers is not in the document]
Comment (ShiriMoran) [reply pkg/ibmvpc/vpc.go:231]: Its not related to this PR. https://github.com/np-guard/vpc-network-config-analyzer/issues/840 opened

File: pkg/commonvpc/vpc.go [hunk 3]
```diff
@@ -226,22 +226,26 @@ func (psr *privateSubnetRule) Deny(isIngress bool) bool {
...
+	case !detail && !psr.Deny(false), !detail && !psr.Deny(true):
+		return fmt.Sprintf("public subnet %s enables connection", psr.subnet.Name())
+	case detail && psr.Deny(false), detail && psr.Deny(true):
+		direction := "outbound"
+		if psr.Deny(true) {
+			direction = "inbound"
+		}
+		return fmt.Sprintf("external %s connection is denied since subnet %s is private\n", direction, psr.subnet.Name())
+	default:
+		return fmt.Sprintf("traffic from %s to %s is allowed, since subnet %s is public\n",
```
[the remark this answers is not in the document]
Comment (ShiriMoran) [reply pkg/commonvpc/vpc.go:246]: When detail is high then there is a differentiation between egress and ingress.
When  it is low the explanation should be short, without the additional egress/ingress details.

[the remark this answers is not in the document]
Comment (ShiriMoran) [reply pkg/commonvpc/vpc.go:244]: However, a bug regarding subnet in path for external existed and was fixed. Does not change this example; another relevant example added

**Review by haim-kermany — APPROVED** [review]

**Comment by ShiriMoran** [comment]
closing since contained in #843

**Pull request closed without merging**